modules/mobile/bin/health_checks/token_fetcher.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+require 'capybara'
+require 'yaml'
+
+class TokenFetcher
+  TOKEN_URL = 'https://va-mobile-cutter.herokuapp.com'
+  USERS_FILE = ENV['STAGING_TEST_USERS']
+
+  attr_reader :token, :session
+
+  def initialize(user_name)
+    set_user(user_name)
+    Capybara.default_max_wait_time = 10
+    @session = Capybara::Session.new(:selenium)
+  end
+
+  def fetch_token
+    @session.visit TOKEN_URL
+    @session.click_link('Please Login')
+    sleep 3 # this sleep appears to be necessary for dealing with the redirect
```

Comment:
@jperk51 the file you commented on was in the wrong directory (i started working on this in another directory and failed to delete some of that work). Capybara is supposed to wait for page load by default, but I wasn't able to get it working without the sleep here. I assume it's because a number of redirects are made, but I'm actually not certain why. I'm definitely open to better solutions.

Reply:
Gotcha, I think this is probably fine overall, but could fail with slow connections is all. In the past I've set up webcrawlers with loops that are basically halting execution until a known element of the next page is detected. Not sure if that's really worth doing here or not.

Reply:
It does sometimes fail on one of the "Accept" buttons not being there but it seems to work otherwise. We can definitely iterate on this. A loop might be better than a flat sleep. It's a good suggestion. I'll give it a try. But my overall understanding is that modern capybara should wait for elements to render. I just didn't see it working in this one spot.